flocker/acceptance/test_api.py

```diff
@@ -0,0 +1,96 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for the control service REST API.
+"""
+
+import socket
+
+from signal import SIGINT
+from os import kill
+from uuid import uuid4
+from json import dumps, loads
+
+from twisted.trial.unittest import TestCase
+
+from treq import get, post, content
+
+from .testtools import get_nodes, _run_SSH
+from ..testtools import loop_until
+
+
+def wait_for_api(hostname):
+    """
+    Wait until REST API is available.
+
+    :return Deferred: Fires when REST API is available.
+    """
+    def api_available():
+        try:
+            s = socket.socket()
+            s.connect((hostname, 4523))
+            return True
+        except socket.error:
+            return False
+    return loop_until(api_available)
+
+
+class DatasetAPITests(TestCase):
+    """
+    Tests for the dataset API.
+    """
+    def test_dataset_creation(self):
+        """
+        A dataset can be created on a specific node.
+        """
+        # This is blocking for now, may as well due this the succinct way:
```

Comment:
Seems like the wrong shortcut to take.  If it's blocking, maybe it shouldn't return a Deferred.  If it returns a Deferred, you should add a callback.  `successResultOf` is not for this.